www/js/lib/zimfile.js

```diff
@@ -223,15 +244,85 @@ define(['xzdec_wrapper', 'zstddec_wrapper', 'util', 'utf8', 'q', 'zimDirEntry',
                 } else {
                     return new Uint8Array(); // unsupported compression type
                 }
-                return decompressor.readSliceSingleThread(blob * 4, 8).then(function (data) {
+                return decompressor.readSliceSingleThread(blob * 4, 8, false).then(function (data) {
                     var blobOffset = readInt(data, 0, 4);
                     var nextBlobOffset = readInt(data, 4, 4);
-                    return decompressor.readSliceSingleThread(blobOffset, nextBlobOffset - blobOffset);
+                    return decompressor.readSliceSingleThread(blobOffset, nextBlobOffset - blobOffset, true);
                 });
             });
         });
     };
 
+    /**
+     * A Directory Listing object
+     * @typedef {Object} DirListing A list of pointers to directory entries (via the URL pointerlist)
+     * @property {String} path The path (url) to the directory entry for the Listing
+     * @property {String} ptrName The name of the pointer to the Listing's data that will be added to the ZIMFile obect
+     * @property {String} countName The name of the key that will contain the number of entries in the Listing, to be added to the ZIMFile object 
+     */
+
+    /**
+     * Read the metadata (archive offset pointer, and number of entiries) of one or more ZIM directory Listings.
+     * This supports reading a subset of user content that might be ordered differently from the main URL pointerlist.
+     * In particular, it supports v1 title pointerlists, which contain articles sorted by title, superseding the article
+     * namespace ('A') in legazy ZIM archives.  
+     * @param {Array<DirListing>} listings An array of DirListing objects (see zimArchive.js for examples)  
+     */
+    ZIMFile.prototype.setListings = function(listings) {
+        // If we are in a legacy ZIM archive, there is nothing further to look up
+        if (this.minorVersion === 0) {
+            console.log('ZIM DirListing version: legacy');
```

Comment:
@mossroy Are you OK with our keeping this `console.log` entry (and the one below line 282) when merging? These will only ever print out a single message when a user first loads a ZIM, and I think it might be quite a useful diagnostic to have in case we notice any problem with production no-namespace ZIMs going forward.

Reply:
OK for me, but use debug level : `console.debug` instead of `console.log`

Reply:
OK, thanks. I wasn't aware of `console.debug()`. Looks useful.